resqpy/grid/create_grid_xml.py

```diff
@@ -0,0 +1,379 @@
+                ccw_values_node.text = rqet.null_xml_text
+
+                grid.model.create_hdf5_dataset_ref(ext_uuid,
+                                                   grid.uuid,
+                                                   'KJI'[axis] + 'Regrid/ChildCellWeights',
+                                                   root = ccw_values_node)
+
+        # todo: handle omit and cell overlap functionality as part of parent window refining or coarsening
+
+    if write_geometry:
+        __write_geometry(ext_uuid, grid, ijk)
+
+    if add_as_part:
+        __add_as_part(add_relationships, ext_uuid, grid, ijk, write_geometry)
+
+    if (write_active and grid.active_property_uuid is not None and
+            grid.model.part(uuid = grid.active_property_uuid) is None):
+        # TODO: replace following with call to rprop.write_hdf5_and_create_xml_for_active_property()
+        active_collection = rprop.PropertyCollection()
+        active_collection.set_support(support = grid)
+        active_collection.create_xml(None,
+                                     None,
+                                     'ACTIVE',
+                                     'active',
+                                     p_uuid = grid.active_property_uuid,
+                                     discrete = True,
+                                     add_min_max = False,
+                                     find_local_property_kinds = True)
+
+    return ijk
+
+
+def __add_as_part(add_relationships, ext_uuid, grid, ijk, write_geometry):
+    grid.model.add_part('obj_IjkGridRepresentation', grid.uuid, ijk)
+    if add_relationships:
+        if grid.stratigraphic_column_rank_uuid is not None and grid.stratigraphic_units is not None:
+            grid.model.create_reciprocal_relationship(ijk, 'destinationObject',
+                                                      grid.model.root_for_uuid(grid.stratigraphic_column_rank_uuid),
+                                                      'sourceObject')
+        if write_geometry:
+            # create 2 way relationship between IjkGrid and Crs
+            grid.model.create_reciprocal_relationship(ijk, 'destinationObject', grid.crs_root, 'sourceObject')
+            # create 2 way relationship between IjkGrid and Ext
+            ext_part = rqet.part_name_for_object('obj_EpcExternalPartReference', ext_uuid, prefixed = False)
+            ext_node = grid.model.root_for_part(ext_part)
+            grid.model.create_reciprocal_relationship(ijk, 'mlToExternalPartProxy', ext_node, 'externalPartProxyToMl')
+        # create relationship with parent grid
+        if grid.parent_window is not None and grid.parent_grid_uuid is not None:
+            grid.model.create_reciprocal_relationship(ijk, 'destinationObject',
+                                                      grid.model.root_for_uuid(grid.parent_grid_uuid), 'sourceObject')
+
+
+def __write_geometry(ext_uuid, grid, ijk):
```

Comment:
This function would probably more accurately be named `__add_geometry_xml() `or something like that.  The original method probably used write_geometry as an argument name to indicate that it should have the same value passed to it as a write hdf5 method, where that argument name is more descriptive of its effect!